### PR TITLE
Pigeonhole_Sort_Using_Kotlin

### LINKED ISSUE #2149: Pigeonhole Sort
- [x] C
- [x] C++
- [x] C# - @Harikrishnan6336
- [ ] CoffeeScript
- [ ] Dart - @heisastark
- [x] Go - @m-code12
- [x] Java
- [ ] JS - @sheetalssr
- [ ] Kotlin - @Anushka118
- [ ] Php
- [x] Python
- [x] Readme
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Anushka118**
Fixes #2149 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Created pigeonhole sort algorithm in Kotlin with sample input and outputs

#### Languages Used:
- Kotlin 

#### Files Added:
- PigeonHoleSort.kt
- [SampleInOutKotlin.txt](https://github.com/jainaman224/Algo_Ds_Notes/files/4303649/SampleInOutKotlin.txt)
-file link:
https://drive.google.com/file/d/1f3s3OmmzHOc2Vwf98q5NsJWFmnipBH_Q/view?usp=sharing

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by jainaman224** [comment]
Fix PR title.

File: Pigeonhole_Sort/PigeonHoleSort.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import java.util.*
+
+fun main(args: Array<String>) {
+
+    var n: Int = Integer.valueOf(readLine())    // taking input for size of the array
+    var arr = arrayOfNulls<Int>(n)              // declaring arr of size n
+    var read= Scanner(System.`in`)
```
Comment (jainaman224) [inline Pigeonhole_Sort/PigeonHoleSort.kt]: Add a space before `=`

File: Pigeonhole_Sort/PigeonHoleSort.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import java.util.*
+
+fun main(args: Array<String>) {
+
+    var n: Int = Integer.valueOf(readLine())    // taking input for size of the array
+    var arr = arrayOfNulls<Int>(n)              // declaring arr of size n
+    var read= Scanner(System.`in`)          
+    for(i in 0..(n-1)){
+        arr[i] = read.nextInt()                 //input value of each array element
```
Comment (jainaman224) [inline Pigeonhole_Sort/PigeonHoleSort.kt]: Add a space after `//`

File: Pigeonhole_Sort/PigeonHoleSort.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Sample Input
+5
+1 5 4 2 0
+Sample Output
+Sorted Array
+0 
+1 
+2 
+4 
+5
```
Comment (jainaman224) [inline Pigeonhole_Sort/PigeonHoleSort.kt]: Please end the comment using `*/`

File: Pigeonhole_Sort/PigeonHoleSort.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    var max = arr[0]                            //max element in the array
+    //finding the minimum and maximum elements in the array
+    for(i in 0..(n-1)){
+        if(arr[i]!! < min!!)
+            min = arr[i]
+        if(arr[i]!! > max!!)
+            max = arr[i]
+    }
+
+    var noOfPigeonHoles = max!! - min!! + 1     //size of the pigeonhole array
```
Comment (jainaman224) [inline Pigeonhole_Sort/PigeonHoleSort.kt]: Can you use `!!` in different way like `!!max`.
It makes program unreadable beyond this point.

**Comment by Anushka118** [comment]
@jainaman224 
I have made the asked changes in the file and also changed the title of the PR. Please review the PR.

**Comment by raghav-dalmia** [comment]
Looks good to me. Squash your commits. I will merge it after that.

**Comment by Anushka118** [comment]
Squashed all the commits. Please merge it.

**Comment by Anushka118** [comment]
@raghav-dalmia please merge the PR as i am not able to work on any other issues

**Comment by Anushka118** [comment]
@jainaman224 @raghav-dalmia please merge my pull request as I have already made the necessary changes and I am not able to work on other issues.
Please consider my request.

File: Pigeonhole_Sort/PigeonHoleSort.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    
+    for(i in 0..(noOfPigeonHoles-1)){
+        while(pigeonHoles[i] > 0){
+            arr[index++] = i + min
+            pigeonHoles[i]--
+        }
+    }
+    
+    // printing the array
+
```
Comment (somya-kapoor) [inline Pigeonhole_Sort/PigeonHoleSort.kt:42]: Remove this line.

File: Pigeonhole_Sort/PigeonHoleSort.kt [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import java.util.*
```
Comment (somya-kapoor) [inline Pigeonhole_Sort/PigeonHoleSort.kt:1]: Please change the file name to Pigeonhole_Sort/Pigeonhole_Sort.kt

File: Pigeonhole_Sort/PigeonHoleSort.kt [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        if(!!arr[i] > !!max)
+            max = arr[i]
+    }
+
+    var noOfPigeonHoles = !!max - !!min + 1     // size of the pigeonhole array
+    var pigeonHoles = IntArray(noOfPigeonHoles) // pigeonhole array
+    Arrays.fill(pigeonHoles,0)                  // initializing all the elements to 0
+
+    // pigeonhole sort algorithm
+
```
Comment (somya-kapoor) [inline Pigeonhole_Sort/PigeonHoleSort.kt:27]: Remove this line

**Comment by Anushka118** [comment]
I am not able to make any changes in this file anymore as I accidentally deleted the branch from which it was created. Is it any other way I can access my file? @somya-kapoor 
I still have the original file with me. 
Also previously one of the mentors, Raghav Dalmia commented that the PR is looking fine and I just had to squash all the comments to merge it, which I did.
Now, there are supposed to be more changes in the file.

**Comment by somya-kapoor** [comment]
Filename should be the same. See other pigeonhole file.

**Comment by Anushka118** [comment]
As stated above I cannot access my file so I am closing this pull request and creating a new one.

**Pull request closed without merging**